### ISSUE TITLE
Map Winston levels to LogDNA levels

### ISSUE THREAD

**Issue by kendallroth** [logdna/logdna-winston#31]
In our project there is a discrepancy between the log levels supported by [`@logdna/logger`](https://github.com/logdna/logger-node#supported-log-levels) and the levels we are using in WinstonJS (we also use `CRITICAL`). We have not found a way to "map" between the Winston log levels (custom or otherwise) to the list of supported LogDNA levels.

For example, a potential workaround would be mapping our WinstonJS `critical` log level to the LogDNA `error` log level (since they do not support `critical`). However, currently this is not possible, so any of our `critical` logs get reported as `info` in LogDNA.

Is there any way to get support for additional log levels, or at least mapping them?

P.S. The [LogDNA docs](https://docs.logdna.com/docs/ingestion#log-level) apparently support more common log levels, but their package does not... _See slightly different ask in [`@logdna/logger`](https://github.com/logdna/logger-node/issues/53)_

**Comment by kendallroth** [comment]
While debugging this I came across a few potential issues. Most notably, this library **only** supports the NPM log levels; however, WinstonJS provides several log level configurations ([docs](https://github.com/winstonjs/winston#logging-levels))! Furthermore, this library does not support custom log levels, as the mapping between WinstonJS and LogDNA levels is hardcoded ([file](https://github.com/logdna/logdna-winston/blob/main/index.js#L8))! This is quite frustrating for developers working with custom log levels, as it means there is no way to pass some log levels through to LogDNA (or map them).

Is there a way that this level mapping could be customizable?

**Comment by kendallroth** [comment]
As a workaround, I created a patch file with `patch-package` to resolve our specific issue. Ideally however, it would be best to provide a custom log level mapping instead of generating patch files.

```patch
diff --git a/node_modules/logdna-winston/index.js b/node_modules/logdna-winston/index.js
index e17031a..81abcf9 100644
--- a/node_modules/logdna-winston/index.js
+++ b/node_modules/logdna-winston/index.js
@@ -6,7 +6,8 @@ const pkg = require('./package.json')
 
 // Convert between Winston levels and @logdna/logger levels
 const levelTranslate = new Map([
-  ['error', 'error']
+  ['critical', 'critical']
+, ['error', 'error']
 , ['warn', 'warn']
 , ['info', 'info']
 , ['http', 'debug']
```

**Comment by kendallroth** [comment]
@darinspivey  Regarding your brief mention of this repository in [Issue #53](https://github.com/logdna/logger-node/issues/53#issuecomment-912555480), will there also be a change to support a custom `Map` (or something) between potentially custom WinstonJS log levels and the supported (and maybe custom) LogDNA levels?

**Comment by darinspivey** [comment]
@kendallroth Yes, once custom levels are supported, this package will get a `major` change to allow the usage of custom log levels.  TBD.

**Comment by darinspivey** [comment]
@kendallroth Please let us know how this change works for you.

**Comment by kendallroth** [comment]
@darinspivey Thank you so much for your work on this; I have tested (albeit not in production yet) and verified that this resolves our use case. Simply being able to use the custom Winston levels along with a quick call to `Object.keys()` is extremely nice!

_P.S. While testing this I did run into an unrelated bug with the transport `level` option (used differently by LogDNA and WinstonJS), but have opened a separate [issue](https://github.com/logdna/logdna-winston/issues/33) for that)_
